syntax = "proto3";

package play;

service Turn {
    rpc SetMove (Step) returns (State) {}
    rpc GetMove (Player) returns (Step) {}
    rpc UpdateNext (State) returns (State) {}
    rpc IsNextPlayer (Player) returns (State) {}
    rpc SetPlayer (Player) returns (State) {}
    rpc GetAIPlayer (State) returns (Player) {}
    rpc HasChosen (State) returns (State) {}
    rpc HasMoved (Player) returns (State) {}
}

message Step {
    uint32 x = 1;
    uint32 y = 2;
    Player player = 3;
}

message Player {
    uint32 color = 1;
}

message State {
    bool status = 1;
}


